cmd/cone/terraform_gen.go

```diff
@@ -0,0 +1,189 @@
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// TODO @anthony: this probably could be cleaned up if we made the template generic
+func getResourceMap(ctx context.Context, c client.C1Client, v *viper.Viper, object string) (map[string]resource.TemplateData, error) {
+	resources := make(map[string]resource.TemplateData)
+	if object == "app" || object == "*" {
+		apps, err := c.ListApps(ctx)
+		if err != nil {
+			return nil, err
+		}
+		for _, app := range apps {
+			tmplData := resource.AppTemplate{App: app}
+			resources[tmplData.GetOutputId()] = tmplData
+		}
+	}
+	if object == "policy" || object == "*" {
+		policies, err := c.ListPolicies(ctx)
+		if err != nil {
+			return nil, err
+		}
+		for _, policy := range policies {
+			tmplData := resource.PolicyTemplate{Policy: policy}
+			resources[tmplData.GetOutputId()] = tmplData
+		}
+	}
+
+	if object == "app_entitlement" || object == "*" {
+		appId := v.GetString(tfAppIdFlag)
+		if appId == "" {
+			return nil, errors.New("app-id flag is required for app_entitlement object")
+		}
+
+		entitlements, err := c.ListEntitlements(ctx, appId)
+		if err != nil {
+			return nil, err
+		}
+		for _, entitlement := range entitlements {
+			tmplData := resource.AppEntitlementTemplate{AppEntitlement: entitlement}
+			resources[tmplData.GetOutputId()] = tmplData
+		}
+	}
+	return resources, nil
+}
+
+func terraformGen(cmd *cobra.Command, args []string) error {
+	ctx, c, v, err := cmdContext(cmd)
+	if err != nil {
+		return err
+	}
+
+	if err := validateArgLenth(2, args, cmd); err != nil {
+		return err
+	}
+
+	object := args[0]
+	if !slices.Contains(objects, object) && object != "*" {
+		return fmt.Errorf("invalid object name, only support %v and * for all", objects)
+	}
+
+	terraformDir := args[1]
+	if _, err := os.Stat(terraformDir); err != nil {
+		return fmt.Errorf("terraform directory %s does not exist", terraformDir)
+	}
+
+	// Turns objects into dataTemplates
+	resources, err := getResourceMap(ctx, c, v, object)
+	if err != nil {
+		return err
+	}
+
+	/* TODO @anthony: this all could be simplier if our terraform provider was better at imports.
+	* Currently imports are not supported for some nested objects, for example, policy.steps gets imported incorrectly.
+	* This way of doing it forces datasources to match resources, which is not ideal.
+	*
+	* For each object, we create a template that will import the datasource then output it.
+	 */
+	outputTemplate, err := resource.ApplyTemplates(maps.Values(resources), resource.DataTemplateString, resource.OutputTemplateString)
+	if err != nil {
+		return err
+	}
+	err = writeToFile(terraformDir+"/cone_temp.tf", outputTemplate)
+	if err != nil {
+		return err
+	}
+
+	// By running the command the output of terraform plan is piped to a text file
+	pterm.Info.Println("Please run this command in the terraform directory:")
+	pterm.Info.Printfln(`touch %s; terraform plan | sed 's/\x1b\[[0-9;]*m//g'> %s`, tempFile, tempFile)
```

Comment:
Can we just do this programatically in Golang instead of having the user run a secondary command?

Reply:
Depends on the users system, if they have a windows machine it will not work, also i found it caused issues when I ran it using go, it is not reliable.